### DISCUSSION TITLE
How is the scenario of updating the "base" branch handled by the tool?

### DISCUSSION THREAD

**Discussion by mkarajohn** [ejoffe/spr#398] [Q&A]
I am in the process of comparing this and Graphite, and one scenario I have not yet had the chance to test, is the following:

>You have some stacks branching off of the base branch, and at the same time the base branch gets updated, for whatever reason.

With Graphite, updating descendant branches is a matter of running a single command.

Is there a way to do something similar here or does one have to manually rebase each descendant branch to base? 

Thank you

**Comment by ejoffe** [comment]
`git spr update` does it all for you

**Reply by mkarajohn** [reply]
Oh, I thought this only pushed local stuff to origin.

So say I have a bunch of stacks, branching off of master, and master gets updated by another contributor.

How would it work? I checkout my local `master`, pull from origin and then run `git spr update`, that's all? All stacks get rebased on top of master?

Thanks

**Reply by ejoffe** [reply]
Yes. And you don't need to pull. 
Just run spr update on your local branch.

**Reply by ejoffe** [reply]
Note that each local branch is a single stack.

**Reply by mkarajohn** [reply]
That's pretty neat! Thank you for clarifying! 

Would you consider updating the "Managing Pull Requests" section of the docs to make this functionality more apparent? The way it's written currently reads as if it only updates the remote based on your local stuff. 

I could try to open a PR for it, but considering I just started using this (plus the fact that you have a much better picture of the internals and how they work, than me), I don't know if I am the appropriate person for this; we'll probably just end up with a back-and-forth in the PR :sweat_smile:

**Reply by mkarajohn** [reply]
> Note that each local branch is a single stack.

YY, I have this in mind. 

So, wait, I would have to run `git spr update` on each local branch/stack or just once on my `master` branch? Now I got confused again and I thought I had it figured out :sweat_smile:

**Reply by ejoffe** [reply]
In spr your local branch is the whole stack of commits, each branch is a stack. 
Normally, I only have a single stack going at a time, I always work on the master branch,  but sometimes when working on two completely different features, I'll start a new branch (stack) for the second feature. Then I'll have my main master branch, plus a second feature branch. Each branch is it's own stack, and you run spr update in that branch to rebase on top of origin and update the pull requests. 
I suggest you get some confidence by using the tool, and once you have experience and feel that the docs can use an update, open a pr and contribute your changes.

**Reply by mkarajohn** [reply]
Yeah, that's probably the best course of action, will do. Thank you for your time!
